app/services/encrypted_attribute.rb

```diff
@@ -0,0 +1,76 @@
+class EncryptedAttribute
+  attr_accessor :user_access_key
+  attr_reader :encrypted, :decrypted
+
+  def self.new_user_access_key(cost: nil, key: nil)
+    env = Figaro.env
+    key ||= env.attribute_encryption_key
+    cost ||= env.attribute_cost
+    UserAccessKey.new(password: key, salt: key, cost: cost)
+  end
+
+  def self.new_from_decrypted(decrypted, user_access_key = new_user_access_key)
+    encryptor = Pii::PasswordEncryptor.new
+    encrypted = encryptor.encrypt(decrypted, user_access_key)
+    new(encrypted, decrypted: decrypted, user_access_key: user_access_key)
+  end
+
+  def initialize(encrypted, decrypted: nil, cost: nil, user_access_key: nil)
+    self.encrypted = encrypted
```

Comment:
I like ivars too. My problem is that reek eventually complains about having too many of them, so the pattern we've evolved is to use explicit methods rather than ivars. A good example is `AttributeAsserter` which accrued so many ivars over time that we switched to explicit method definitions via `attr_writer` and `attr_reader`.